olp-cpp-sdk-authentication/test/unit/ArcGisTestUtils.cpp

```diff
@@ -70,36 +55,24 @@ class ArcGisTestUtils::Impl {
 
   virtual ~Impl() = default;
 
-  bool getAccessToken(ArcGisUser& user);
+  bool getAccessToken(std::shared_ptr<olp::http::Network> network,
+                      const olp::http::NetworkSettings& network_settings,
+                      ArcGisUser& user);
 
  private:
-  ScopedNetworkPtr getScopedNetwork();
   std::shared_ptr<std::vector<unsigned char> > generateClientBody();
-
- private:
-  std::weak_ptr<ScopedNetwork> networkPtr;
-  std::mutex networkPtrLock;
 };
 
-ArcGisTestUtils::Impl::ScopedNetworkPtr
-ArcGisTestUtils::Impl::getScopedNetwork() {
-  lock_guard<mutex> lock(networkPtrLock);
-  auto netPtr = networkPtr.lock();
-  if (netPtr) return netPtr;
-
-  auto result = make_shared<ScopedNetwork>();
-  networkPtr = result;
-  return result;
-}
-
-bool ArcGisTestUtils::Impl::getAccessToken(ArcGisUser& user) {
-  NetworkRequest request(ARCGIS_URL, 0, ::NetworkRequest::PriorityDefault,
-                         ::NetworkRequest::HttpVerb::POST);
-
-  request.SetContent(generateClientBody());
-  request.AddHeader("content-type", "application/x-www-form-urlencoded");
+bool ArcGisTestUtils::Impl::getAccessToken(
+    std::shared_ptr<olp::http::Network> network,
```

Comment:
Pretty expensive a copy each time for a requests.
If it is not async or captured then prefer to pass it as reference:
```suggestion
    olp::http::Network& network,
```